### PR TITLE
UNOMI-352 Implement event type registry for GraphQL

### PR DESCRIPTION
**Pull request by pmi**

### CODE REVIEW

File: api/src/main/java/org/apache/unomi/api/services/EventTypeRegistry.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.unomi.api.services;
+
+import org.apache.unomi.api.EventType;
+
+import java.util.Collection;
+
+public interface EventTypeRegistry {
```
Comment (sergehuber) [inline api/src/main/java/org/apache/unomi/api/services/EventTypeRegistry.java:24]: Please add basic documentation on this class

File: services/src/main/java/org/apache/unomi/services/impl/definitions/DefinitionsServiceImpl.java [hunk 2]
```diff
@@ -59,7 +59,17 @@
 
     private BundleContext bundleContext;
     public DefinitionsServiceImpl() {
-
+        // let's add the built-in value types
```
Comment (sergehuber) [inline services/src/main/java/org/apache/unomi/services/impl/definitions/DefinitionsServiceImpl.java:62]: I wonder if we should add all the built-in value types programmatically like this or move them all to JSON files. But having a mixture of this doesn't seem coherent.

File: services/src/main/java/org/apache/unomi/services/impl/definitions/DefinitionsServiceImpl.java [hunk 3]
```diff
@@ -160,7 +170,7 @@ private void processBundleStop(BundleContext bundleContext) {
         if (bundleContext == null) {
             return;
         }
-        List<PluginType> types = pluginTypes.get(bundleContext.getBundle().getBundleId());
+        List<PluginType> types = pluginTypes.remove(bundleContext.getBundle().getBundleId());
```
Comment (sergehuber) [inline services/src/main/java/org/apache/unomi/services/impl/definitions/DefinitionsServiceImpl.java:173]: Is this a bug fix ?

File: services/src/main/java/org/apache/unomi/services/impl/events/EventServiceImpl.java [hunk 4]
```diff
@@ -211,10 +226,61 @@ private void getEventProperties(Map<String, Map<String, Object>> mappings, List<
...
+        }
+        return properties;
+    }
+
+    private ValueType mappingTypeToValueType(String mappingType) {
+        if ("text".equals(mappingType)) {
+            return definitionsService.getValueType("string");
+        } else if ("date".equals(mappingType)) {
+            return definitionsService.getValueType("date");
+        } else if ("long".equals(mappingType)) {
```
Comment (sergehuber) [inline services/src/main/java/org/apache/unomi/services/impl/events/EventServiceImpl.java:266]: I'm not sure whether we should translate long to integer or introduce a "long" in Unomi. I've been playing with the second idea.

File: services/src/main/resources/META-INF/cxs/events/common.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
+{
+  "propertyTypes": [
+    {
+      "itemId": "id",
+      "type": "id"
+    },
+    {
+      "itemId": "cdp_source"
```
Comment (sergehuber) [inline services/src/main/resources/META-INF/cxs/events/common.json:8]: Should we really have the cdp fields here ? I thought we would have the Event.class fields here and then create a layer to translate them to cdp in the GraphQL code which would seem more coherent.

File: services/src/main/java/org/apache/unomi/services/impl/definitions/DefinitionsServiceImpl.java (hunk 3 above)
Comment (pmi) [reply services/src/main/java/org/apache/unomi/services/impl/definitions/DefinitionsServiceImpl.java:173]: That was done by you in UNOMI-286 that I based my work on, so I can only guess :)

File: services/src/main/java/org/apache/unomi/services/impl/definitions/DefinitionsServiceImpl.java (hunk 2 above)
Comment (pmi) [reply services/src/main/java/org/apache/unomi/services/impl/definitions/DefinitionsServiceImpl.java:62]: That came from UNOMI-286 that I based my work on, but I will remove that, as we have set property type defined in json. Don't think we need _unknown_ though.

File: services/src/main/java/org/apache/unomi/services/impl/events/EventServiceImpl.java (hunk 4 above)
Comment (pmi) [reply services/src/main/java/org/apache/unomi/services/impl/events/EventServiceImpl.java:266]: Yeah, but maybe it's better to do it as a separate task ?

File: services/src/main/resources/META-INF/cxs/events/common.json (hunk 5 above)
Comment (pmi) [reply services/src/main/resources/META-INF/cxs/events/common.json]: Removed common.json completely

File: api/src/main/java/org/apache/unomi/api/services/EventTypeRegistry.java (hunk 1 above)
Comment (pmi) [reply api/src/main/java/org/apache/unomi/api/services/EventTypeRegistry.java:27]: Done

File: api/src/main/java/org/apache/unomi/api/services/EventService.java [hunk 6]
```diff
@@ -82,9 +79,19 @@
      * Retrieves the list of available event properties.
      *
      * @return a list of available event properties
+     * Retrieves an event type
```
[the remark this answers is not in the document]
Comment (pmi) [reply api/src/main/java/org/apache/unomi/api/services/EventService.java]: Fixed

File: services/src/main/java/org/apache/unomi/services/impl/definitions/DefinitionsServiceImpl.java (hunk 2 above)
Comment (sergehuber) [reply services/src/main/java/org/apache/unomi/services/impl/definitions/DefinitionsServiceImpl.java]: Ok yes unknown is not needed let's get rid of it.

File: services/src/main/java/org/apache/unomi/services/impl/definitions/DefinitionsServiceImpl.java (hunk 3 above)
Comment (sergehuber) [reply services/src/main/java/org/apache/unomi/services/impl/definitions/DefinitionsServiceImpl.java:172]: ok sorry didn't realize that :)

File: services/src/main/java/org/apache/unomi/services/impl/events/EventServiceImpl.java (hunk 4 above)
Comment (sergehuber) [reply services/src/main/java/org/apache/unomi/services/impl/events/EventServiceImpl.java:266]: agreed but we will need to remember it somehow

**Comment by sergehuber** [comment]
Thanks for the PR, I just merged it !

**Pull request merged**